### PR TITLE
Enable all supported cipher suites on Java 7

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

Also see https://github.com/gradle/gradle-private/issues/1341

Some weak cipher algorithm might be disabled by default, which might cause SSL handshake throw unexpected exception. This PR enables all supported cipher suites for Java 7 to avoid such exceptions.

### CODE REVIEW

File: subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java [hunk 1]
```diff
@@ -72,30 +73,47 @@
...
     }
 
-    private final HttpSettings httpSettings;
+    private static String[] configureCipherSuites() {
+        String cipherSuites = System.getProperty(HTTPS_CIPHER_SUITES);
+        if (cipherSuites != null) {
+            return cipherSuites.split(",");
+        } else if (JavaVersion.current().isJava7()) {
+            try {
+                return SSLContext.getDefault().getSocketFactory().getSupportedCipherSuites();
```
Comment (wolfs) [inline subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientConfigurer.java]: ❌ It feels very wrong to activate weak cipher suites for everyone when downloading artifacts. The whole point of TLS 1.2 is to make things more secure. Using a weak cypher over TLS 1.2 makes no sense IMO. JCenter needs to be fixed not our code.

**Comment by wolfs** [comment]
Here is some info from the internal issue:

On June 24th jfrog.org removed support of TLSv1.0/1.1 so all clients have to use TLSv1.2 now. However, JDK7's TLSv1.2 ciphers are not compatible with jfrog.org.

**Comment by blindpirate** [comment]
More precisely, according to the [doc](https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLSocketFactory.html#getDefaultCipherSuites--) the algorithms which are not enabled by default are not "weak", they just "do not meet quality of service requirements". For example, [ssllab](https://www.ssllabs.com/ssltest/analyze.html?d=repo.jfrog.org&s=34.226.44.194) says `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256` is not weak, but on my local machine, this algorithm is not enabled by default (`getDefaultCipherSuites()` doesn't contain it but `getSupportedCipherSuites()` does contain it).

supported by `jfrog.org` are:

```
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)   ECDH secp256r1 (eq. 3072 bits RSA)   FS	128
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xc027)   ECDH secp256r1 (eq. 3072 bits RSA)   FS	128
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)   ECDH secp256r1 (eq. 3072 bits RSA)   FS	256
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xc028)   ECDH secp256r1 (eq. 3072 bits RSA)   FS	256
TLS_RSA_WITH_AES_128_GCM_SHA256 (0x9c)   WEAK	128
TLS_RSA_WITH_AES_128_CBC_SHA256 (0x3c)   WEAK	128
TLS_RSA_WITH_AES_256_GCM_SHA384 (0x9d)   WEAK	256
TLS_RSA_WITH_AES_256_CBC_SHA256 (0x3d)   WEAK	256
```

**Comment by ljacomet** [comment]
I agree with @wolfs here. If the JFrog setup forces us to enable less secure cipher suites in Java 7, then something is busted with their setup.

**Comment by blindpirate** [comment]
Closed due to the change on jfrog's side, thank you @ljacomet @wolfs !

**Pull request closed without merging**